network/src/sync/miner.rs

```diff
@@ -45,20 +43,24 @@ impl MinerInstance {
     /// Spawns a new miner on a new thread using MinerInstance parameters.
     /// Once a block is found, A block message is sent to all peers.
     /// Calling this function multiple times will spawn additional listeners on separate threads.
-    pub async fn spawn(self) -> Result<task::JoinHandle<()>, ConsensusError> {
-        let local_address = self.node.local_address().unwrap();
-        info!("Initializing Aleo miner - Your miner address is {}", self.miner_address);
-        let miner = MineContext::prepare(
-            self.miner_address.clone(),
-            Arc::clone(&self.node.expect_sync().consensus),
-        )
-        .await?;
-        info!("Miner instantiated; starting to mine blocks");
-
-        let mut mining_failure_count = 0;
-        let mining_failure_threshold = 10;
-
-        Ok(task::spawn_blocking(move || {
```

Comment:
why do we want to reduce this thread into smaller threads?

Reply:
Answering the previous questions on more fine-grained calls to `spawn_blocking` here; we've discussed this once briefly during the community call: blocking tasks should generally not be long-lived, and are expected to be used for operations that finish on their own ([tokio docs](https://docs.rs/tokio/1.10.1/tokio/index.html#cpu-bound-tasks-and-blocking-code)); other than that, a lot of the code underneath `mine_block` is `async`-dense and having `block_on` in some of the underlying operations makes them likely candidates for future issues in tests/refactorings.

TL;DR: while the current implementation does work, it's a bit "hacky" in terms of `tokio`'s design and could cause issues in the future (either "self-inflicted" or when `tokio` decides to enforce some of their documented behavior in a stricter fashion).